ruby/src/IceRuby/Communicator.cpp

```diff
@@ -155,20 +155,13 @@ IceRuby_initialize(int argc, VALUE* argv, VALUE /*self*/)
 
         if (!NIL_P(initData))
         {
-            volatile VALUE properties = callRuby(rb_iv_get, initData, "@properties");
-            volatile VALUE logger = callRuby(rb_iv_get, initData, "@logger");
-            volatile VALUE initDataSliceLoader = callRuby(rb_iv_get, initData, "@sliceLoader");
-
+            VALUE properties = callRuby(rb_iv_get, initData, "@properties");
```

Comment:
`volatile` here is not required. This code just gets a reference to the properties ruby object pass by the client initData, and unwraps it to get the underlying C++ properties.